### ISSUE TITLE
Maybe it is possible to eval network with BN layers when batchsize>1

### ISSUE THREAD

**Issue by Naruto-Sasuke** [junyanz/pytorch-CycleGAN-and-pix2pix#893]
For net with IN layers, it should be fine when batchsize is larger than 1 when testing.

For net with BN layers, can this also be possible if replacing the BN with IN layers by assigning weights  from corresponding BN layers to those IN layers ? Or just 'Adding a net option' to BN layer, making it perfoms instance normalization (I do current BN does not support it...).

As sometimes, the testing set are very large, and batch-testing mode should benefit a lot.

**Comment by junyanz** [comment]
For IN, it is possible for any batchSize during test time. For BN, if you do not use running mean/std, it should be fine for any batchSize. Converting BN to IN is possible but it seems to be nontrivial.

**Comment by XiaotianM** [comment]
> For IN, it is possible for any batchSize during test time. For BN, if you do not use running mean/std, it should be fine for any batchSize. Converting BN to IN is possible but it seems to be nontrivial.

how to eval Pix2pix with BN when batchsize=1, I find it will also run mean/std during inference

**Comment by XiaotianM** [comment]
If I use net.eval(), the eval result is not same as the results saved in training

**Comment by junyanz** [comment]
During training, your results are produced by running mean/std. Therefore, it is not the same.

**Comment by XiaotianM** [comment]
> During training, your results are produced by running mean/std. Therefore, it is not the same.

but I find whether use eval.() or not , the results can not be good as training's. So how can I reproduce the good results like training's ?

**Comment by junyanz** [comment]
Did you use the same model? Your final saved model might be different from the model during training.

**Comment by XiaotianM** [comment]
> Did you use the same model? Your final saved model might be different from the model during training.

Thanks for your reply
